implementation/jwt-auth/src/main/java/io/smallrye/jwt/auth/principal/DefaultJWTParser.java

```diff
@@ -179,9 +179,9 @@ private JWTAuthContextInfo copyAuthContextInfo() {
 
     private void setSignatureAlgorithmIfNeeded(JWTAuthContextInfo newAuthContextInfo, String algoStart,
             SignatureAlgorithm newAlgo) {
-        SignatureAlgorithm algo = newAuthContextInfo.getSignatureAlgorithm();
-        if (algo == null || !algo.getAlgorithm().startsWith(algoStart)) {
-            newAuthContextInfo.setSignatureAlgorithm(newAlgo);
+        Set<SignatureAlgorithm> algo = newAuthContextInfo.getSignatureAlgorithm();
+        if (algo == null || !algo.stream().anyMatch(s -> s.getAlgorithm().startsWith(algoStart))) {
```

Comment:
Can `algo` ever be null? It would at least be `Set.of(SignatureAlgorithm.RS256)` if not set, correct?